sbom/generator/generator.go

```diff
@@ -22,7 +23,13 @@ func NewBom(report *reporter.Report) ([]*sbom.Sbom, error) {
 	return GenerateBom(report)
 }
 
+func GenerateBomV2(r *reporter.Report) []*sbom.Sbom {
```

Comment:
why do we need that? Can we not just use GenerateBom and return nil as error?

Reply:
I wanted to avoid forcing any upstream repos that use this to migrate since the signature changes. We could also change `GenerateBom`'s signature to never return an error too, I think this cleans up the usage for callers here as they don't have to handle an error

Reply:
Agreed, I will change this